scipy/stats/_continuous_distns.py

```diff
@@ -3528,6 +3530,47 @@ def _ppf(self, q, a, b):
 
 johnsonsu = johnsonsu_gen(name='johnsonsu')
 
+class johnsonsl_gen(rv_continuous):
+    r"""A Johnson SL continuous random variable.
+    %(before_notes)s
+
+    See Also
+    --------
+    johnsonsb
+    johnsonsu
+
+    Notes
+    -----
+    The probability density funcition for a `johnsonsl` is::
+        johnsonsl.pdf(x, a, b) = b/x* phi(a + b * ln (x))
+    for x>0 where ``phi`` is the normal pdf.
+
+    %(after_notes)s
+
+    .. versionadded:: 1.3.0
+
+    %(example)s
+    """
+    def _argcheck(self, a, b):
+        return a==a
+
+    def _pdf(self, x, a, b):
+        #negatives = x<=0
```

Comment:
if you want to define the Distribution on [0, inf), set `a = 0` when you define `johnsonsl = johnsonsl_gen(a=0, name='johnsonsl')`

Reply:
will do thanks